tests/beaker_tests/cisco_acl/test_ace.rb

```diff
@@ -0,0 +1,222 @@
+    'fragments'       => '',
+  },
+}
+
+tests['create_ipv4_seq_10'] = {
+  title_pattern:  'ipv4 beaker_1 10',
+  manifest_props: "
+    'action'        => 'permit',
+    'proto'         => 'tcp',
+    'src_addr'      => '1.2.3.4 2.3.4.5',
+    'src_port'      => 'eq 40',
+    'dst_addr'      => '9.9.0.4/32',
+    'dst_port'      => 'range 32 56',
+  ",
+  resource_props: "
+    'action'        => 'permit',
+    'proto'         => 'tcp',
+    'src_addr'      => '1.2.3.4 2.3.4.5',
+    'src_port'      => 'eq 40',
+    'dst_addr'      => '9.9.0.4/32',
+    'dst_port'      => 'range 32 56',
+  ",
+}
+
+tests['destroy_ipv4_seq_10'] = {
+  title_pattern:  'ipv4 beaker_1 10',
+  manifest_props: "
+    'ensure'        => :absent
+    'action'        => 'permit',
+    'proto'         => 'tcp',
+    'src_addr'      => '1.2.3.4 2.3.4.5',
+    'src_port'      => 'eq 40',
+    'dst_addr'      => '9.9.0.4/32',
+    'dst_port'      => 'range 32 56',
+  ",
+  resource_props: "
+    'action'        => 'permit',
+    'proto'         => 'tcp',
+    'src_addr'      => '1.2.3.4 2.3.4.5',
+    'src_port'      => 'eq 40',
+    'dst_addr'      => '9.9.0.4/32',
+    'dst_port'      => 'range 32 56',
+  ",
+}
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH + 'resource cisco_ace'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_ace(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    manifest = tests[id][:manifest_props_absent]
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_ace :: title_pattern:\n" +
+               tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_ace { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_ace(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_ace(tests, id)
+
+  test_harness_common(tests, id)
+  # add case
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+  # node_feature_cleanup(agent, 'acl')
```

Comment:
Still need to address this #121 comment:
  https://github.com/cisco/cisco-network-puppet-module/pull/121/files#diff-ac20693befc9cf1b0135cf1ba18823a0R204